src/Workspaces/CSharp/Portable/CodeGeneration/MethodGenerator.cs

```diff
@@ -130,6 +167,25 @@ internal static class MethodGenerator
             return methodDeclaration;
         }
 
+        private static LocalFunctionStatementSyntax UseExpressionBodyIfDesired(
+            Workspace workspace, LocalFunctionStatementSyntax localMethodDeclaration, ParseOptions options)
+        {
+            if (localMethodDeclaration.ExpressionBody == null)
+            {
+                var expressionBodyPreference = workspace.Options.GetOption(CSharpCodeStyleOptions.PreferExpressionBodiedMethods).Value;
+                if (localMethodDeclaration.Body.TryConvertToArrowExpressionBody(
+                        localMethodDeclaration.Kind(), options, expressionBodyPreference,
+                        out var expressionBody, out var semicolonToken))
+                {
+                    return localMethodDeclaration.WithBody(null)
+                                                 .WithExpressionBody(expressionBody)
+                                                 .WithSemicolonToken(semicolonToken);
+                }
```

Comment:
are there tests for this?

Reply:
Yep, there are some tests which go through this codepath.